[Shruthi] iP

Shruthi0310 opened the pull request:
# Duke

> "Out of the mountain of despair, a stone of hope"  – Martin Luther King, Jr.

Duke helps you remember things you need to do. It's,
* easy to use
* fast
* text-based

All you need to do is,
1. download it from [here](https://github.com/Shruthi0310/ip)
1. run it in your computer
1. add your tasks
1. let it manage your tasks for you 😃 

*And it's **FREE**!!!*

Features:
- [x] Manage tasks
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java! Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        new Duke("duke.txt").run();
    }
}
```

Commit:
Add Gradle support

Shruthi0310 committed:
Level-1

Shruthi0310 committed:
Level-2

Shruthi0310 committed:
Level-3

Shruthi0310 committed:
Level-4

Shruthi0310 committed:
A-TextUiTesting

Shruthi0310 committed:
Level-5

Shruthi0310 committed:
Level-6

Shruthi0310 committed:
Add save to hard disk feature

Shruthi0310 committed:
Merge branch 'branch-Level-7'

Shruthi0310 committed:
Add date and time parser

Shruthi0310 committed:
Update time parser

Shruthi0310 committed:
Update date/time parser

Shruthi0310 committed:
Refactor code to extract classes

Shruthi0310 committed:
Divide classes into packages

Shruthi0310 committed:
Store dates as localDate instead of String

Shruthi0310 committed:
Add Junit tests

Shruthi0310 committed:
Package the App as a JAR file

Shruthi0310 committed:
Tweak the code to comply with coding standard

Shruthi0310 committed:
Add find matching task feature

Shruthi0310 committed:
Add JavaDoc comments

Shruthi0310 committed:
Merge branch 'branch-A-JavaDoc'

Shruthi0310 committed:
Merge branch 'branch-A-CodingStandard'

Shruthi0310 committed:
Merge branch 'branch-Level-9'

Tanishq4331 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,181 @@
+     *
+     * @return length of list
+     */
+    public int listLength() {
```
This method name could be further improved to use a verb as the name instead. Maybe something like `getListLength`?

Shruthi0310 replied:
Yep! Good point!

Tanishq4331 commented on src/main/java/duke/exception/NoCommandException.java:
```diff
@@ -0,0 +1,7 @@
+package duke.exception;
```
I like how you created multiple exceptions for different situations.

Shruthi0310 replied:
Thanks!

Tanishq4331 commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,247 @@
+import duke.exception.NoCommandException;
+import duke.exception.NoDescriptionAndTimeException;
+import duke.exception.NoDescriptionException;
+import duke.exception.NoTimeException;
```
Perhaps you could add line breaks between different types of imports to improve readability. For instance
```java
import duke.exception.DukeException;
import duke.exception.InvalidTaskDeletionException;
import duke.exception.InvalidTaskDoneException;
import duke.exception.NoCommandException;
import duke.exception.NoDescriptionAndTimeException;
import duke.exception.NoDescriptionException;
import duke.exception.NoTimeException;

import duke.task.TaskList;

import java.text.ParseException;
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
```

Shruthi0310 replied:
Yea that seems much easier to read

Tanishq4331 commented on src/test/java/duke/util/ParserTest.java:
```diff
@@ -0,0 +1,174 @@
+import java.util.ArrayList;
+import org.junit.jupiter.api.Test;
+
+public class ParserTest {
```
I like how comprehensive these unit tests are.

Shruthi0310 replied:
Thank you!!

Tanishq4331 reviewed (commented):
The coding standards have been followed very well. I had a hard time finding violations. 😃

Shruthi0310 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Shruthi0310 commented:
> The coding standards have been followed very well. I had a hard time finding violations. 😃

Shruthi0310 committed:
Add GUI to Duke

Shruthi0310 committed:
Merge branch 'branch-Level-10'

juliussneezer04 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,68 @@
+    /**
+     * Constructor for this task.
+     *
+     * @param description Details of the task
```
Remember to add punctuations after header comment statements too! (Like full-stops here)

Shruthi0310 replied:
Right I will add them!

juliussneezer04 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,175 @@
+import java.time.format.DateTimeParseException;
+
+import java.util.ArrayList;
+
```
I like how you have added whitespace between import statements of different packages! Adds to the readability 👍

Shruthi0310 replied:
Thanks!

juliussneezer04 reviewed (commented):
Great code! 😄 I guess the previous reviewer did most of my work for me! Looks great now :)

Shruthi0310 committed:
Add assert feature

There are important assumptions that should hold at
various points in the code.

The use of the assert feature helps detect and correct
programming errors.

Let's insert assert feature where there are invariants
in the methods.

Using assert is preferable over exceptions because
it helps detect errors within the code itself.

Shruthi0310 committed:
Refactor code to improve code quality.

The code contains a few long methods and has some
arrow-head style code, and violates a few other code
quality standards.

Refactoring the code to follow the code quality standards
make the code easier to read by other programmers.

Let's change the code like shorterning the long methods and
removing other violations of code quality standards.

Shruthi0310 committed:
Merge pull request #1 from Shruthi0310/branch-A-Assertions

Add assert feature

Shruthi0310 committed:
Merge branch 'master' into branch-A-CodeQuality

Shruthi0310 committed:
Merge pull request #2 from Shruthi0310/branch-A-CodeQuality

Refactor code to improve code quality.

Shruthi0310 committed:
Add archive feature

Users have to delete all current tasks to start with
a clean slate and will not be able to look at past tasks.

The archive feature is more user-friendly as users can
start with a clean slate by typing a single command and also
keep track of past completed tasks.

Let's update the taskList class to have archive feature that
clears previous tasks and saves it in another file.

Using archive is better than delete since users can remove
unnecessary tasks but still look at them while for delete
they will never be able to see it again.

Shruthi0310 committed:
Refactor code to use more OOP for commands

Shruthi0310 committed:
Update ui for list command

Shruthi0310 committed:
Add a User Guide

Shruthi0310 committed:
Add representative screenshot of product

Shruthi0310 committed:
Update Junit tests

Shruthi0310 committed:
Improve coding standards

Shruthi0310 committed:
Update event and deadline commands

Shruthi0310 committed:
Update Ui message format

Shruthi0310 committed:
Update product screenshot

Shruthi0310 committed:
Update JavaDoc comments

Shruthi0310 committed:
Update build.gradle

Shruthi0310 committed:
Update README.md

Shruthi0310 committed:
Update UserGuide.md
